pkg/storage/unified/resource/index_test.go

```diff
@@ -110,3 +103,38 @@ func namespaces() []string {
 	}
 	return ns
 }
+
+func newTestIndex(t *testing.T) *Index {
+	tracingCfg := tracing.NewEmptyTracingConfig()
+	trace, err := tracing.ProvideService(tracingCfg)
+	assertNil(t, err)
+
+	return &Index{
+		tracer: trace,
+		shards: make(map[string]Shard),
+		log:    log.New("unifiedstorage.search.index"),
+		opts: Opts{
+			ListLimit: 5000,
+			Workers:   10,
+			BatchSize: 1000,
+		},
+	}
+}
+
+func assertNil(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func assertCount(t *testing.T, index *Index, expected uint64) {
+	total, err := index.Count()
+	assertNil(t, err)
+	assert.Equal(t, expected, total)
```

Comment:
I wouldn't put this into a dedicated function, it's only used two times. Even if used multiple times, having it directly in the testing code makes it much easier to understand the code.

I would also change the API to something like that:

```golang
total, err := index.Count()
require.NoError(t, err)
require.Equal(t, expected, total)
```

Reply:
it's only used 2 times but we will be adding many more test scenarios